### ISSUE TITLE
neat_basic - resolver closes not properly

### ISSUE THREAD

**Issue by weinrank** [NEAT-project/neat#13]
Valgrind reports a memory leak for the neat_basic app:
http://buildbot.nplab.de:28010/builders/ubuntu-14.04/builds/32/steps/shell_1/logs/stdio

In the neat basic_example we stop the event_loop after reading the server's answer:
https://github.com/NEAT-project/neat/blob/master/samples/neat_basic.c#L42

Then we call `neat_free_flow` and `neat_free_ctx`:
https://github.com/NEAT-project/neat/blob/master/samples/neat_basic.c#L107-L108

tl;dr; ... `neat_resolver_idle_cb` is never called because the event_loop has been stopped and memory not freed.
